src/components/common/DocumentDelivery.vue

```diff
@@ -0,0 +1,101 @@
+      </v-col>
+      <v-col :cols="9">
+        <v-card-text id="office-email" class="pa-0">{{registeredOfficeEmail || 'Not entered'}}</v-card-text>
+      </v-col>
+    </v-row>
+    <v-row class="py-3" no-gutters>
+      <v-col :cols="3">
+        <label class="font-weight-bold" :class="{ 'error-text': invalidSection }">Completing Party</label>
+      </v-col>
+      <v-col cols="9" class="pr-1" v-if="editableCompletingParty">
+        <v-text-field
+          v-model="optionalEmail"
+          id="optionalEmail"
+          class="text-input-field"
+          filled
+          label="Client Email Address (Optional)"
+          hint="Example: name@email.com"
+          persistent-hint
+          validate-on-blur
+          :rules="entityEmailRules"
+        />
+      </v-col>
+      <v-col :cols="9" v-else>
+        <v-card-text id="completing-party-email" class="pa-0">{{userEmail || 'Not entered'}}</v-card-text>
+      </v-col>
+    </v-row>
+    <v-row class="py-3" no-gutters v-if="showCustodianEmail">
+      <v-col :cols="3">
+        <label class="font-weight-bold">Custodian of Records</label>
+      </v-col>
+      <v-col :cols="9">
+        <v-card-text id="custodian-email" class="pa-0">{{custodianEmail || 'Not entered'}}</v-card-text>
+      </v-col>
+    </v-row>
+  </v-card>
+</template>
+
+<script lang="ts">
+// Libraries
+import { Component, Emit, Prop, Vue, Watch } from 'vue-property-decorator'
+
+@Component({})
+export default class DocumentDelivery extends Vue {
+  @Prop({ default: null }) readonly registeredOfficeEmail: string
+  @Prop({ default: null }) readonly custodianEmail: string
+  @Prop({ default: null }) readonly userEmail: string
+  @Prop({ default: null }) readonly documentOptionalEmail: string
+
+  @Prop({ default: false }) readonly editableCompletingParty: boolean
+  @Prop({ default: false }) readonly showCustodianEmail: boolean
+  @Prop({ default: false }) readonly invalidSection: boolean
+
+  // Local properties
+  private optionalEmail: string = ''
+
+  private entityEmailRules = [
+    (v: string) => !/^\s/g.test(v) || 'Invalid spaces', // leading spaces
+    (v: string) => !/\s$/g.test(v) || 'Invalid spaces', // trailing spaces
+    (v: string) => this.validateEmailFormat(v) || 'Enter valid email address'
+  ]
+
+  mounted () {
+    this.optionalEmail = this.documentOptionalEmail
+  }
+
+  private validateEmailFormat (value: string): boolean {
+    // allow empty as the email is optional
+    if (!value) {
+      return true
+    } else {
+      const VALID_FORMAT = new RegExp(/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/)
+      return VALID_FORMAT.test(value)
+    }
+  }
+
+  @Watch('optionalEmail')
+  private async onOptionalEmailChanged (val: string): Promise<void> {
+    if (this.validateEmailFormat(val)) {
+      await this.emitOptionalEmail()
```

Comment:
I like the new prop-event interface in this component 👍 

I am wondering whether these emitters (lines 92 and 97 below) are truly async? (When we worked on NR UI there was a question about whether some store methods were async.) I believe we have never awaited them in our other projects and things have worked correctly. If the emitters are not async then THIS watcher doesn't need to be async either.